redaxo/src/core/lib/var/var.php

```diff
@@ -125,6 +129,26 @@ public static function parse($content, $env = null, $context = null, $contextDat
         return $content;
     }
 
+    /**
+     * @return Iterator<self>
+     */
+    public static function varsIterator(string $content): Iterator
+    {
+        $matches = self::getMatches($content);
```

Comment:
Ich glaube, dass die Methode aktuell nicht wirklich sauber die rex-vars liefert. die parse() methode arbeitet ja token-basiert. so werden zum beispiel rex-vars in php-kommentaren ignoriert. der iterator feuert den regex aber einfach auf den gesamten content ab, also keine token-basierte betrachtung.

Bei der token-basierten Betrachtung wird auch berücksichtigt, in was für Anführungszeichen die Rex-var vorkommt, was Auswirkungen auf das Escaping der Anführungszeichen innerhalb der Rex-Var hat.

```php
$foo = 'REX_MY_VAR[foo=\'bar\']';
```

Reply:
ich bin hier nicht wei genug im thema. wäre super wenn du es dir anschauen könntest...

Reply:
ich hab nochmal drauf geschaut, aber habs noch nicht überblickt. hättest du zeit mal zu schauen, wie es am besten passen würde?

Reply:
Ich habe jetzt nochmal einen Blick drauf geworfen. Ich denke, dass man die Klasse dafür umbauen müsste, da die parse-Methode, die die Tokenzerlegung macht, auch direkt die Ersetzung macht. Insgesamt ist das keine einfache Sache, denke ich. Gerade weil je nach Token-Kontext die Rex-Vars auch unterschiedliche Escapings enthalten können (wie oben angedeutet).

Ich möchte meine Energie eigentlich mehr dafür verwenden, die Abschaffung der Rex-Vars weiter vorzubereiten. Also weitere PHP-Alternativen schaffen. Die Leute jetzt schon dazu bringen, weniger die Rex-Vars zu nutzen (dort wo schon möglich), etc.